src/main.ts

```diff
@@ -177,6 +192,7 @@ function animate(): void {
     // ################################
     // TODO @Shinigami92 2022-09-21: Filtering the enemies in each render cycle is a performance issue
     const enemies = scene.children.filter(isEnemy);
+    const minerals = scene.children.filter(isMineral);
```

Comment:
Minerals could be detected on game reset as they will never change position and will game is running they will never be new minerals added dynamically